Can't run benchmarks

Seems to have something to do with features. Or maybe the problem is just that `ConsensusProgram` is `pub(crate)`? Making it `pub` yields new warnings, though.

```sh
$ cargo criterion --bench consensus --all-features
   Compiling neptune-cash v0.0.11 (/home/thv/repos/neptune-core)
error[E0603]: trait `ConsensusProgram` is private
  --> benches/consensus.rs:27:66
   |
27 |     use neptune_cash::models::proof_abstractions::tasm::program::ConsensusProgram;
   |                                                                  ^^^^^^^^^^^^^^^^ private trait
   |
note: the trait `ConsensusProgram` is defined here
  --> /home/thv/repos/neptune-core/src/models/proof_abstractions/tasm/program.rs:24:1
   |
24 | pub(crate) trait ConsensusProgram
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

```bash
$ cargo criterion --bench consensus
error: target `consensus` in package `neptune-cash` requires the features: `arbitrary-impls`
Consider enabling them by passing, e.g., `--features="arbitrary-impls"`
Error: 'cargo bench' returned an error (exit status: 101); unable to continue.
```
```bash
$ cargo criterion --bench consensus --features="arbitrary-impls"
   Compiling neptune-cash v0.0.11 (/home/thv/repos/neptune-core)
error[E0433]: failed to resolve: use of undeclared crate or module `proptest`
   --> src/models/blockchain/transaction/primitive_witness.rs:421:9
    |
421 |     use proptest::arbitrary::Arbitrary;
    |         ^^^^^^^^ use of undeclared crate or module `proptest`

error[E0433]: failed to resolve: use of undeclared crate or module `proptest`
   --> src/models/blockchain/transaction/primitive_witness.rs:422:9
    |
422 |     use proptest::collection::vec;
    |         ^^^^^^^^ use of undeclared crate or module `proptest`
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -36,6 +36,9 @@ jobs:
       - name: Run clippy
         run: cargo clippy --all-targets -- -D warnings
 
+      - name: Build benches
+        run: cargo build --benches
+
       - name: Run tests without benches
         run: >
           cargo test

--- a/src/models/proof_abstractions/tasm/program.rs
+++ b/src/models/proof_abstractions/tasm/program.rs
@@ -21,7 +21,10 @@ pub enum ConsensusError {
 
 /// A `ConsensusProgram` represents the logic subprogram for transaction or
 /// block validity.
-pub(crate) trait ConsensusProgram
+///
+/// This trait is required for benchmarks, but is not part of the public API.
+#[doc(hidden)]
+pub trait ConsensusProgram
 where
     Self: RefUnwindSafe + std::fmt::Debug,
 {
@@ -51,6 +54,12 @@ where
     ///
     /// This method is a thin wrapper around [`prove_consensus_program`], which
     /// does the same but for arbitrary programs.
+    //
+    // The entire trait is only `pub` to facilitate benchmarks; it is not part of
+    // the public API. The suppressed lints below are not nice, but I don't know
+    // how else to make it work.
+    #[allow(async_fn_in_trait)]
+    #[allow(private_interfaces)]
     async fn prove(
         &self,
         claim: Claim,

